backend/app/modules/computation/Clusterizer.scala

```diff
@@ -0,0 +1,236 @@
+package modules.computation
+
+import java.awt.image.BufferedImage
+import java.io.File
+import javax.imageio.ImageIO
+
+import scala.annotation.tailrec
+import scala.collection.mutable.ListBuffer
+
+
+/**
+  * Utility class generating a cluster of points from an Image.
+  *
+  * This class read an RGB image, get channels to clusterize (e.g. Red only or
+  * average between multiple channels) and, using a treshold value, create
+  * clusters of points.
+  *
+  * As an example, suppose we have the following red channel :
+  *     255 234 180 120 97  123 12
+  *     201 243 177 12  18  129 60
+  *     120 169 100 243 230 112 190
+  *     12  54  91  251 200 100 0
+  *     0   15  71  161 182 190 100
+  *
+  * Using the a treshold value of 200, we will set to one all points above the
+  * treshold to 1 and the other ones to 0:
+  *     1 1 0 0 0 0 0
+  *     1 1 0 0 0 0 0
+  *     0 0 0 1 1 0 0
+  *     0 0 0 1 1 0 0
+  *     0 0 0 0 0 0 0
+  *
+  * On this matrice we can see two different groups of pixels. This class will
+  * generate two cluster, one in (x: 1/2, y: 1/2) and the other one in (x: 9/2,
+  * y: 9/2), both with a radius of the maximum distance from the baricentre,
+  * i.e. sqrt(1/2) ~= 0.7.
+  */
+class Clusterizer(imageFile: File) {
+
+  /**
+    * Lazy load of the image pixels
+    */
+  val image: BufferedImage = ImageIO.read(imageFile)
+
+  /**
+    * Image width.
+    */
+  lazy val width: Int = image.getWidth
+
+  /**
+    * Image height.
+    */
+  lazy val height: Int = image.getHeight
+
+  /**
```

Comment:
Add a todo, for creating clusters AFTER the load of the class, so the user can modify the threshold value.